src/neo-vm/ExecutionEngine.cs

```diff
@@ -307,147 +229,132 @@ private bool ExecuteInstruction()
                             context.EvaluationStack.Set(0, xn);
                             break;
                         }
                     case OpCode.XTUCK:
                         {
-                            if (!context.EvaluationStack.TryPop(out PrimitiveType item_n))
-                                return false;
+                            if (!TryPop(out PrimitiveType item_n)) return false;
                             int n = (int)item_n.ToBigInteger();
                             if (n <= 0) return false;
-                            context.EvaluationStack.Insert(n, context.EvaluationStack.Peek());
+                            context.EvaluationStack.Insert(n, Peek());
                             break;
                         }
                     case OpCode.DEPTH:
                         {
-                            context.EvaluationStack.Push(context.EvaluationStack.Count);
-                            if (!CheckStackSize(true)) return false;
+                            Push(context.EvaluationStack.Count);
                             break;
                         }
                     case OpCode.DROP:
                         {
-                            context.EvaluationStack.Pop();
-                            CheckStackSize(false, -1);
+                            if (!TryPop(out StackItem _)) return false;
                             break;
                         }
                     case OpCode.DUP:
                         {
-                            context.EvaluationStack.Push(context.EvaluationStack.Peek());
-                            if (!CheckStackSize(true)) return false;
+                            Push(Peek());
                             break;
                         }
                     case OpCode.NIP:
                         {
-                            context.EvaluationStack.Remove(1);
-                            CheckStackSize(false, -1);
+                            if (!context.EvaluationStack.TryRemove(1, out StackItem _)) return false;
                             break;
                         }
                     case OpCode.OVER:
                         {
-                            context.EvaluationStack.Push(context.EvaluationStack.Peek(1));
-                            if (!CheckStackSize(true)) return false;
+                            Push(Peek(1));
                             break;
                         }
                     case OpCode.PICK:
                         {
-                            if (!context.EvaluationStack.TryPop(out PrimitiveType item_n))
-                                return false;
+                            if (!TryPop(out PrimitiveType item_n)) return false;
                             int n = (int)item_n.ToBigInteger();
                             if (n < 0) return false;
-                            context.EvaluationStack.Push(context.EvaluationStack.Peek(n));
+                            Push(Peek(n));
```

Comment:
As I said on https://github.com/neo-project/neo-vm/pull/232/files#r352554553 , here we have computed already the current context, if we use the new method, it will calculate it again twice